uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-request-action-dropdown.js

```diff
@@ -0,0 +1,172 @@
+/* eslint-env amd */
+
+define([
+  'common/lodash',
+  'common/moment',
+  'leave-absences/shared/modules/components',
+  'common/services/hr-settings'
+], function (_, moment, components) {
+  components.component('leaveRequestActionDropdown', {
+    bindings: {
+      request: '<',
+      requestStatuses: '<',
+      absenceTypes: '<',
+      /**
+       * Role is not a permission level in this case.
+       * For example Manager can act as Staff
+       * and Admin can act as either Manager or Staff.
+       */
+      role: '<'
+    },
+    templateUrl: ['shared-settings', function (sharedSettings) {
+      return sharedSettings.sharedPathTpl + 'components/leave-request-action-dropdown.html';
+    }],
+    controllerAs: 'vm',
+    controller: ['$log', '$q', '$rootScope', 'OptionGroup', 'dialog', 'shared-settings', controller]
+  });
+
+  function controller ($log, $q, $rootScope, OptionGroup, dialog, sharedSettings) {
+    $log.debug('Component: leave-request-action-dropdown');
+
+    var vm = this;
+
+    vm.actions = [];
+    vm.actionLabels = {
+      edit: 'Edit',
+      cancel: 'Cancel',
+      delete: 'Delete',
+      view: 'View',
+      respond: 'Respond',
+      approve: 'Approve',
+      reject: 'Reject'
+    };
+
+    /**
+     * Performs an action on a given leave request
+     *
+     * @param {LeaveRequestInstance} leaveRequest
+     * @param {string} action
+     */
+    vm.act = function (action) {
```

Comment:
@AkA84 this is not an abbreviation. This is a verb *to act*. But OK, I changed it to *action*.